src/components/UpgradeButton.js

```diff
@@ -1,22 +1,63 @@
-import React from "react";
+import React, { useState } from "react";
 import PropTypes from "prop-types";
 
-const UpgradeButton = ({ type }) => {
+import Button from "@material-ui/core/Button";
+import Menu from "@material-ui/core/Menu";
+import MenuItem from "@material-ui/core/MenuItem";
+
+const UpgradeButton = ({
+  type,
+  upgrades,
+  upgradeFunction,
+  unitKey,
+  upgradeKey
+}) => {
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [selectedUpgrade, setSelectedUpgrade] = useState(undefined);
+
+  const handleClick = event => {
```

Comment:
`handleClick` isn't particularly descriptive. Something more accurate to the actual behavior would be `openUpgradeMenu` or something along those lines. It also helps readability when you pass this function as an event handler.

Before:
`<Button variant="contained" onClick={handleClick}>`

After:
`<Button variant="contained" onClick={openUpgradeMenu}>`

You instantly have a much better idea of what the button is actually going to do and saves you from having to drill into the function to see what's going on. Additionally, just looking at the code inside of this it's not immediately obvious what would happen unless you see how anchorEl is used, so having a well named function also serves as a way of documenting the functions intent.